core/src/main/kotlin/com/alecstrong/sql/psi/core/sql.bnf

```diff
@@ -231,9 +231,7 @@ expr ::= ( raise_expr
 other_expr ::= extension_expr
 extension_expr ::= NULL
 extension_stmt ::= NULL
-multi_column_expr ::= values_expression [ ( LT | LTE | GT | GTE ) values_expression ] {
-  pin = 1
```

Comment:
Honestly, I don't understand pinning at all. What is the benefit?

Reply:
yes, its important for error handling, so that if you mistype an expression it parses enough to tell you what you should have typed next.

Reply:
But only error handling? If I add the pin again, it will stop at `multi_column_expr` and does not try to parse the next rule: `paren_expr`.

Reply:
should this rule just not have the optional? 

```
values_expression ( LT | LTE | GT | GTE ) values_expression
```

Reply:
then it would correctly use paren expr

Reply:
You are right, sql-psi works, but only without the pin. With the pin, it fails to use paren_expr.

Reply:
pin should be on 2 for this one

Reply:
so it only parses as a multi column expr when it gets to the comparison operator